zerver/tests/test_subs.py

```diff
@@ -4732,26 +4732,48 @@ def test_bulk_subscribe_MIT(self) -> None:
             stream.is_in_zephyr_realm = True
             stream.save()
 
-        # Make sure Zephyr mirroring realms such as MIT do not get
-        # any tornado subscription events
-        with self.capture_send_event_calls(expected_num_events=0):
-            with self.assert_database_query_count(5):
+        with self.capture_send_event_calls(expected_num_events=41):
+            with self.assert_database_query_count(52):
                 self.common_subscribe_to_streams(
                     mit_user,
                     stream_names,
                     dict(principals=orjson.dumps([mit_user.id]).decode()),
                     subdomain="zephyr",
-                    allow_fail=True,
                 )
 
-        with self.capture_send_event_calls(expected_num_events=0):
+        with self.capture_send_event_calls(expected_num_events=1):
             bulk_remove_subscriptions(
                 realm,
                 users=[mit_user],
                 streams=streams,
                 acting_user=None,
             )
 
+    def test_subscribe_others_to_public_stream_in_zephyr_realm(self) -> None:
+        """
+        Users cannot be subscribed to public streams by other users in zephyr realm.
+        """
+        starnine = self.mit_user("starnine")
+        espuser = self.mit_user("espuser")
+
+        realm = get_realm("zephyr")
+        stream = self.make_stream("stream_1", realm=realm)
+        stream.is_in_zephyr_realm = True
+        stream.save()
+
+        result = self.common_subscribe_to_streams(
+            starnine,
+            ["stream_1"],
+            dict(principals=orjson.dumps([starnine.id, espuser.id]).decode()),
+            subdomain="zephyr",
+            allow_fail=True,
+        )
+        self.assert_json_error(
+            result,
+            "You can only invite other Zephyr mirroring users to private streams.",
+            status_code=400,
```

Comment:
I think this should be in same commit because removing this causes the commit making changes in backend  to fail some tests as there would not be 100% test coverage. This test was added to fix the failing backend tests.

Reply:
You can probably add this test in a commit before this one mentioning that the case of subsribing others to public streams was not tested. Since in the `test_bulk_subscribe_MIT` test, the user subscribes themselves and not anyone else.